### ISSUE TITLE
Обрезается контекстное меню и плейлист

### ISSUE THREAD

**Issue by meloman27** [umnik1/yaamp#60]
В версии 0.0.2 такое происходит:
![yaamp](https://github.com/umnik1/yaamp/assets/46092799/0e32e8ca-7ffa-4ff1-9aeb-d1a7ac469018)

**Comment by basterbac** [comment]
Опытным путем выяснил, что такое происходит, если отключить отображение эквалайзера. 
Помимо того, что обрезается контекстное меню, пропадает возможность растянуть окно плейлиста, также обрезается.

**Comment by meloman27** [comment]
Да, поэтому в названии темы указал, что плейлист тоже обрезается

**Comment by derunix** [comment]
У меня тоже снизу обрезается

**Comment by dimk-o** [comment]
на версии 0.0.7 Windows 11 происходит ровно такая-же ситуация(

<img width="234" height="612" alt="Image" src="https://github.com/user-attachments/assets/83840c09-2966-47ac-a2a7-1b3d918ca9b4" />

**Comment by sfreest** [comment]
Как workaround - можно растягивать основное окно, и тогда меню не будет обрезаться.
<img width="1149" height="877" alt="Image" src="https://github.com/user-attachments/assets/7d68e39a-606e-4149-a153-e9d18f2cf714" />

Для мейнтейнера немного анализа причины от ChatGPT (PR делать не буду, потому что не шарю во фронтовом стеке):

Проблема воспроизводится на текущем `main` / `v0.0.7`.

Причина — ошибка при выборе направления раскрытия вложенного меню:

- При нехватке места справа код безусловно добавляет класс [`[submenu-left]`](https://github.com/umnik1/yaamp/blob/97211aac4570b95b88794084684d70241a1da74a/src/node/preload.js#L165-L181).
- CSS раскрывает такое подменю целиком влево: [`context-menu.css:L96-L101`](https://github.com/umnik1/yaamp/blob/97211aac4570b95b88794084684d70241a1da74a/webamp/packages/webamp/css/context-menu.css#L96-L101).
- При этом размер Electron-окна рассчитывается только по правой границе (`maxRight`); отрицательная левая координата подменю не учитывается: [`preload.js:L150-L161`](https://github.com/umnik1/yaamp/blob/97211aac4570b95b88794084684d70241a1da74a/src/node/preload.js#L150-L161)  [`preload.js:L183-L202`](https://github.com/umnik1/yaamp/blob/97211aac4570b95b88794084684d70241a1da74a/src/node/preload.js#L183-L202).
- Вручную растянуть системное окно нельзя, поскольку для `BrowserWindow` установлено [`[resizable: false]`](https://github.com/umnik1/yaamp/blob/97211aac4570b95b88794084684d70241a1da74a/main.js#L38-L47).

В результате длинное подменю, например «Станции», не помещается справа, открывается влево с отрицательной `x`-координатой и обрезается границей `BrowserWindow`.

Без изменения кода есть только обходной путь: увеличить рабочую ширину, перетащив одно из внутренних окон вправо, включив Milkdrop/«О треке» сбоку либо растянув Playlist по горизонтали. Положение сохраняется в пользовательском `settings.json`. Отдельной настройки ширины меню или пользовательского CSS в проекте нет.

Минимальная правка — открывать подменю влево только тогда, когда оно действительно помещается слева:

```js
const fitsRight =
  parentRect.right + submenuWidth <= viewportWidth + 1;
const fitsLeft =
  parentRect.left - submenuWidth >= 0;

parentEl.classList.toggle(
  "submenu-left",
  !fitsRight && fitsLeft
);
```

Если подменю не помещается ни с одной стороны, его лучше оставить справа: существующий `updateElectronWindowBounds()` после появления подменю сможет увеличить окно по `submenuRect.right`.

**Comment by bstalgoteam** [comment]
<img width="1400" height="1183" alt="Image" src="https://github.com/user-attachments/assets/ef3570b0-5137-4f07-9ffc-239ad974e0cb" />

В моем случае обрезка происходит по всем элементам плеера

**Comment by kandreev** [comment]
У меня выглядит так будто по левому краю обрезается.

<img width="999" height="797" alt="Image" src="https://github.com/user-attachments/assets/ed1128c0-9be7-4c88-8e7f-c938bd38d857" />
